service/history/workflow_task_handler_callbacks_update_test.go

```diff
@@ -0,0 +1,410 @@
+
+	mockTxProcessor := queues.NewMockQueue(s.controller)
+	mockTimerProcessor := queues.NewMockQueue(s.controller)
+	mockVisibilityProcessor := queues.NewMockQueue(s.controller)
+	mockArchivalProcessor := queues.NewMockQueue(s.controller)
+	mockMemoryScheduledQueue := queues.NewMockQueue(s.controller)
+	mockTxProcessor.EXPECT().Category().Return(tasks.CategoryTransfer).AnyTimes()
+	mockTimerProcessor.EXPECT().Category().Return(tasks.CategoryTimer).AnyTimes()
+	mockVisibilityProcessor.EXPECT().Category().Return(tasks.CategoryVisibility).AnyTimes()
+	mockArchivalProcessor.EXPECT().Category().Return(tasks.CategoryArchival).AnyTimes()
+	mockMemoryScheduledQueue.EXPECT().Category().Return(tasks.CategoryMemoryTimer).AnyTimes()
+	mockTxProcessor.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+	mockTimerProcessor.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+	mockVisibilityProcessor.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+	mockArchivalProcessor.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+	mockMemoryScheduledQueue.EXPECT().NotifyNewTasks(gomock.Any()).AnyTimes()
+
+	h := &historyEngineImpl{
+		currentClusterName: mockShard.GetClusterMetadata().GetCurrentClusterName(),
+		shardContext:       mockShard,
+		clusterMetadata:    mockClusterMetadata,
+		executionManager:   s.mockExecutionMgr,
+		logger:             s.logger,
+		throttledLogger:    s.logger,
+		metricsHandler:     metrics.NoopMetricsHandler,
+		tokenSerializer:    common.NewProtoTaskTokenSerializer(),
+		config:             config,
+		timeSource:         mockShard.GetTimeSource(),
+		eventNotifier:      events.NewNotifier(clock.NewRealTimeSource(), metrics.NoopMetricsHandler, func(namespace.ID, string) int32 { return 1 }),
+		queueProcessors: map[tasks.Category]queues.Queue{
+			mockArchivalProcessor.Category():    mockArchivalProcessor,
+			mockTxProcessor.Category():          mockTxProcessor,
+			mockTimerProcessor.Category():       mockTimerProcessor,
+			mockVisibilityProcessor.Category():  mockVisibilityProcessor,
+			mockMemoryScheduledQueue.Category(): mockMemoryScheduledQueue,
+		},
+		workflowConsistencyChecker: api.NewWorkflowConsistencyChecker(mockShard, s.workflowCache),
+	}
+	mockShard.SetEngineForTesting(h)
+
+	s.workflowTaskHandlerCallback = newWorkflowTaskHandlerCallback(h)
+}
+
+func (s *WorkflowTaskHandlerCallbacksUpdateSuite) TearDownTest() {
+}
+
+func (s *WorkflowTaskHandlerCallbacksUpdateSuite) TestAcceptComplete() {
+
+	tv := testvars.New(s.T().Name())
+
+	s.mockNamespaceCache.EXPECT().GetNamespaceByID(tv.NamespaceID()).Return(tv.Namespace(), nil).AnyTimes()
+
+	_, serializedTaskToken := s.createStartedWorkflow(tv)
+
+	writtenHistoryCh := s.writtenHistoryCh(1)
+
+	updRequestMsg, upd := s.createSentUpdate(tv, "1")
+	s.NotNil(upd)
+
+	_, err := s.workflowTaskHandlerCallback.handleWorkflowTaskCompleted(context.Background(), &historyservice.RespondWorkflowTaskCompletedRequest{
```

Comment:
is this _action_ part of the test? may be we can bring `s.EqualHistoryEvents` closer to action?

Reply:
yes, this is **action**. Asserts on update status are also important.